internal/service/s3/bucket_lifecycle_configuration.go

```diff
@@ -164,9 +164,9 @@ func ResourceBucketLifecycleConfiguration() *schema.Resource {
 							Elem: &schema.Resource{
 								Schema: map[string]*schema.Schema{
 									"newer_noncurrent_versions": {
-										Type:         schema.TypeInt,
+										Type:         nullable.TypeNullableInt,
```

Comment:
Only potential "problem"  I see is that this could change user expectations when using the value as an output variable. Previously was `null` or `0`, now it will be the empty String when not configured